alerting/alerter.go

```diff
@@ -1,9 +1,25 @@
 package alerting
 
 import (
-	"github.com/datajet-io/peekaboo/services"
+	"github.com/uber-go/zap"
 )
 
+// NewAlerterClient is a helper function to create native alerter clients
+func NewAlerterClient(name string, config map[string]interface{}) *Alerter {
+	var client Alerter
+
+	switch config["type"] {
+	case "pagerduty":
+		client = NewPagerdutyClient(config["integration_key"].(string))
+	}
+	return &client
+}
+
+// Alerter generic interface type
 type Alerter interface {
-	TriggerAlert(service services.Service, alert Alert, details map[string]interface{}) error
+	AlerterCreate(serviceName string, logger zap.Logger, alert *Alert, details map[string]interface{}) error
+	AlerterResolve(serviceName string, logger zap.Logger, alert *Alert, details map[string]interface{}) error
 }
+
+//Alerters list of alerter handlers to use
+type Alerters map[string]*Alerter
```

Comment:
so its not a list as it is stated in a comment, it is actually a map. AlerterMap then?